src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/DefaultPartitionedRateLimiter.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Threading.RateLimiting
+{
+    internal sealed class DefaultPartitionedRateLimiter<TResource, TKey> : PartitionedRateLimiter<TResource> where TKey : notnull
+    {
+        private readonly Func<TResource, RateLimitPartition<TKey>> _partitioner;
+        private static TimeSpan _idleTimeLimit = TimeSpan.FromSeconds(10);
+
+        // TODO: Look at ConcurrentDictionary to try and avoid a global lock
```

Comment:
TODO for this PR or a follow-up?

Reply:
Follow-ups, this code isn't new, the first commit contained a bunch of code organization.